### ISSUE TITLE
plus the new categories base on the last.pt 

### ISSUE THREAD

**Issue by fhm524017277** [ultralytics/ultralytics#18033]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

How to use yolov8’s last.pt file to add new categories and it will keep the high map？

**Comment by UltralyticsAssistant** [comment]
👋 Hello @fhm524017277, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for guidance, specifically the sections about [Custom Training](https://docs.ultralytics.com/guides/model-training-tips/) which might help you better understand how to add new categories while utilizing your `last.pt` file.

If this is a custom training ❓ Question, please provide as much detail as possible about your process, including:
- Steps you’ve taken to add new categories
- Any dataset examples or information about your dataset
- Relevant training logs or results
This will allow us to assist you more effectively.

For real-time learning and sharing, join our Ultralytics community on [Discord](https://discord.com/invite/ultralytics) 🎧, ask detailed questions on [Discourse](https://community.ultralytics.com), or connect with like-minded users on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Before further troubleshooting, make sure you're working with the latest `ultralytics` package. Update it along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment using [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

This ensures you're benefiting from the latest features and fixes.

## Supported Environments  

YOLO models can be fully utilized in the following environments:

- **Notebooks** (with free GPU):  
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
  
- **Cloud Services**:  
  - Follow the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) for Google Cloud.  
  - Or use AWS with guidance from the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).

- **Docker Containers**:  
  A [Docker Image](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) is available to simplify environment setup.  
  <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## CI Status  

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  

If this badge is green, it means that all of our [Continuous Integration (CI)](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. These tests validate the core functionalities on macOS, Windows, and Ubuntu across various modes and tasks.

---

This is an automated response to help guide you towards the right resources 🌟. An Ultralytics engineer will review and assist you further as soon as possible! 😊

**Comment by Y-T-G** [comment]
You need to train on both the original and new classes.

**Comment by Y-T-G** [comment]
Did you follow this?

https://www.reddit.com/r/Ultralytics/comments/1eolwl8/the_correct_way_to_train_from_a_previously/

You also need to set the optimizer.

**Comment by fhm524017277** [comment]
> Did you follow this?
> 
> https://www.reddit.com/r/Ultralytics/comments/1eolwl8/the_correct_way_to_train_from_a_previously/
> 
> You also need to set the optimizer.
yes! well,i will try once again and set the optimizer,thanks for your reminder

**Comment by glenn-jocher** [comment]
You're welcome! If you still encounter issues, ensure you're using appropriate learning rates and consider adjusting hyperparameters like warmup epochs. Feel free to share specific details if further assistance is needed.

**Comment by fhm524017277** [comment]
> Did you follow this?
> 
> https://www.reddit.com/r/Ultralytics/comments/1eolwl8/the_correct_way_to_train_from_a_previously/
> 
> You also need to set the optimizer.

I try to retrain and i set warm_epocher,lr0 and optimizer,but it's precision and map50 still lower 10%than the first time training,i guess it can not be avoided?

**Comment by glenn-jocher** [comment]
@fhm524017277 the performance drop might be due to differences in data distribution or insufficient fine-tuning. To mitigate this, ensure the optimizer is correctly configured, use a smaller learning rate, and train for more epochs while monitoring overfitting. You can also try strategies like hyperparameter tuning or augmenting your dataset.

**Comment by fhm524017277** [comment]
> @fhm524017277 the performance drop might be due to differences in data distribution or insufficient fine-tuning. To mitigate this, ensure the optimizer is correctly configured, use a smaller learning rate, and train for more epochs while monitoring overfitting. You can also try strategies like hyperparameter tuning or augmenting your dataset.

now，i can join the new data to the model, and it'performance is good,so i try the 
incremental learning,and the training time continue to increase with datasets increasing,i try to freeze the backbone layer ,but it still cost too much time,it's not even as good as retraining，how can i for lessing the traing time

**Comment by glenn-jocher** [comment]
To reduce training time, you can try using a smaller dataset subset for incremental learning, lower the number of epochs, or adjust the batch size to maximize GPU utilization. Freezing more layers or using mixed precision training (`--amp`) can also help. However, for larger updates, retraining from scratch might still be more efficient.

**Comment by fhm524017277** [comment]
> To reduce training time, you can try using a smaller dataset subset for incremental learning, lower the number of epochs, or adjust the batch size to maximize GPU utilization. Freezing more layers or using mixed precision training (`--amp`) can also help. However, for larger updates, retraining from scratch might still be more efficient.
But i set 10patience for earlystoping,i want my model can have a good perfortmance,and i set the lr0 and warm_epoches,freeze=10,all of programs i set   seems make the training time longer. so i use the default.yaml ,jthe training time just slower than the imcremental traing a little bit

**Comment by glenn-jocher** [comment]
It seems the incremental training setup is not providing significant time savings compared to default training. This can happen due to overhead from freezing layers or specific configurations. If retraining with the default settings yields comparable or better performance with only slightly longer training time, it may be more efficient to proceed with retraining rather than incremental updates. For further optimization, you can explore caching (`cache=True`) or using mixed precision (`amp=True`) to speed up training.
